crates/stark-backend/src/keygen/mod.rs

```diff
@@ -195,11 +196,33 @@ impl<'a, SC: StarkGenericConfig> MultiStarkKeygenBuilder<'a, SC> {
             threshold: log_up_security_params.max_interaction_count,
         });
 
+        let pre_vk: MultiStarkVerifyingKey0<SC> = MultiStarkVerifyingKey0 {
+            per_air: pk_per_air.iter().map(|pk| pk.vk.clone()).collect(),
+            trace_height_constraints: trace_height_constraints.clone(),
+            log_up_pow_bits: log_up_security_params.log_up_pow_bits,
+        };
+        // To protect against weak Fiat-Shamir, we hash the "pre"-verifying key and include it in the
+        // final verifying key. This just needs to commit to the verifying key and does not need to be
+        // verified by the verifier, so we just use bincode to serialize it.
+        let vk_bytes = bitcode::serialize(&pre_vk).unwrap();
+        tracing::info!("pre-vkey: {} bytes", vk_bytes.len());
+        // Purely to get type compatibility and convenience, we hash using pcs.commit as a single row
+        let vk_as_row = RowMajorMatrix::new_row(
+            vk_bytes
+                .into_iter()
+                .map(Val::<SC>::from_canonical_u8)
```

Comment:
Observation: The commitment can be made a bit faster if we pack three bytes into the field.

Reply:
yea... a bit hacky so I skipped it